### PR TITLE
Add numeric string completion suggestions as int suggestions

### PR DESCRIPTION
**Pull request by emilyy-dev**
Opted to use Guava's `Ints.tryParse(String)` because it's faster than checking with a regex pattern like how [Sponge does it](https://github.com/SpongePowered/Sponge/blob/ffb81f51a97c06b8bf6692e41aebb8bcd38b1195/src/main/java/org/spongepowered/common/util/CommandUtil.java#L62-L70).

### CODE REVIEW

**Comment by Machine-Maker** [comment]
Change looks good, but what effect does this have on players? Like where does IntegerSuggestion differ from the normal Suggestion?

**Comment by jpenilla** [comment]
My previous testing implies this makes no difference: https://github.com/PaperMC/Paper/pull/5927#issuecomment-870471306

**Comment by Owen1212055** [comment]
Can confirm this doesn’t have any change, it’s in the end send as strings to the client.

**Comment by emilyy-dev** [comment]
> Can confirm this doesn’t have any change, it’s in the end send as strings to the client.

It is, but it's ordered server-side differently.

> My previous testing implies this makes no difference: [#5927 (comment)](https://github.com/PaperMC/Paper/pull/5927#issuecomment-870471306)

Unlike that other PR, this one does not introduce any new API, there is no way for the plugin dev to discern between "This is going to be an int suggestion" vs "This is going to be a text suggestion", and instead it just mimics the ([linked](https://github.com/SpongePowered/Sponge/blob/ef080f509ef42dc0447a6147b01bc1401ddd707b/src/main/java/org/spongepowered/common/util/CommandUtil.java#L64-L72)) behavior presented by Sponge: if the text represents an integer, then it *is*  suggested as such. This has the effect of being ordered numerically as shown here:
```java
event.completions(List.of(
    completion("10"),
    completion("str20"),
    completion("20"),
    completion("1"),
    completion("str10"),
    completion("2"),
    completion("str2"),
    completion("str1")
));
```
### Without this patch:
image


### With this patch:
image

___
Also yes, I just realized this should've been a fixup commit instead :laughing: will fix that now!

**Review by Owen1212055 — APPROVED** [review]
Tested, looks good!
Before:
image

After:
image

File: patches/server/0624-Allow-for-Component-suggestion-tooltips-in-AsyncTabC.patch [hunk 1]
```diff
@@ -35,10 +35,11 @@ index e3d0c7f357a579811de1b55d8fde018aa55fea47..acbaba8750c373b90f6af0fb592f7226
 -            completions.forEach(builder::suggest);
 +            final com.mojang.brigadier.suggestion.SuggestionsBuilder builder = builder0.createOffset(builder0.getInput().lastIndexOf(' ') + 1);
 +            completions.forEach(completion -> {
-+                if (completion.tooltip() == null) {
-+                    builder.suggest(completion.suggestion());
++                final Integer intSuggestion = com.google.common.primitives.Ints.tryParse(completion.suggestion());
```
Comment (jpenilla) [inline patches/server/0624-Allow-for-Component-suggestion-tooltips-in-AsyncTabC.patch]: would prefer a local variable for the result of `completion.tooltip() == null ? null : PaperAdventure.asVanilla(completion.tooltip())`

**Review by jpenilla — DISMISSED** [review]

File: patches/server/0624-Allow-for-Component-suggestion-tooltips-in-AsyncTabC.patch (hunk 1 above)
Comment (emilyy-dev) [reply patches/server/0624-Allow-for-Component-suggestion-tooltips-in-AsyncTabC.patch]: Looking back at it it seems like `PaperAdventure.asVanilla(n.k.a.t.Component)` returns null if the argument is null. I was wondering if you'd still prefer the tooltip component in a local var or just call the function directly when calling `builder.suggest`?

Comment (jpenilla) [reply patches/server/0624-Allow-for-Component-suggestion-tooltips-in-AsyncTabC.patch]: calling the function directly is fine

**Comment by emilyy-dev** [comment]
rebased

**Comment by emilyy-dev** [comment]
rebased

**Review by electronicboy — APPROVED** [review]

**Comment by emilyy-dev** [comment]
rebased

**Pull request merged**